nrf_security/src/mbedtls/cc310/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+get_target_property(NRF_CC310_MBEDCRYPTO_INCLUDE_PATH mbedcrypto_cc310_imported BASE_INCLUDE_PATH)
+
+#
+# Creating a "noglue" version of the mbedcrypto CC310 libray.
+# This library will be compiled with configuration made for the backend
+# and will rely on header-files from the backend to override vanilla
+# context sizes
+#
+# Note that some files are added from mbedcrypto vanilla to ensure that
+# the noglue library is amended with missing symbols
+#
+# The noglue library will not be linked with directly. It is to be combined with
+# the CC310 binary library to create mbedcrypto_cc310 (naming neccesary for
+# renaming in glue )
+#
+add_library(mbedcrypto_cc310_noglue STATIC
+  ${ZEPHYR_BASE}/misc/empty_file.c
+)
+
+target_sources_ifdef(CONFIG_CC310_MBEDTLS_ECJPAKE_C
+  mbedcrypto_cc310_noglue PRIVATE ${ARM_MBEDTLS_PATH}/library/ecjpake.c
+)
+
+target_include_directories(mbedcrypto_cc310_noglue PRIVATE
+  ${NRF_CC310_MBEDCRYPTO_INCLUDE_PATH}
+  ${common_includes}
+  ${config_include}
+)
+
+if(TARGET platform_cc310)
+  target_include_directories(mbedcrypto_cc310_noglue PRIVATE
+    ${mbedcrypto_glue_include_path}/threading
+    ${NRF_PLATFORM_CC310_INCLUDE_PATH}
+  )
+endif()
+
+target_compile_options(mbedcrypto_cc310_noglue PRIVATE ${TOOLCHAIN_C_FLAGS})
+target_ld_options(mbedcrypto_cc310_noglue PRIVATE ${TOOLCHAIN_LD_FLAGS})
+
+target_compile_definitions(mbedcrypto_cc310_noglue PRIVATE
+  -DMBEDTLS_CONFIG_FILE="nrf-config.h"
+)
+
+nrf_security_debug_list_target_files(mbedcrypto_cc310_noglue)
+
+set(cc310_libs
+  mbedcrypto_cc310_imported
+  mbedcrypto_cc310_noglue
+)
+
+combine_archives(mbedcrypto_cc310 "${cc310_libs}")
+symbol_strip_func(cc310)
+
+#
+# Copy alternate files to generated folder (if enabled in backend)
+#
+configure_file_ifdef(CONFIG_CC310_MBEDTLS_AES_C
+  ${NRF_CC310_MBEDCRYPTO_INCLUDE_PATH}/aes_alt.h
+  ${generated_include_path}/aes_alt.h
+  COPYONLY)
+
+configure_file_ifdef(CONFIG_CC310_MBEDTLS_CCM_C
```

Comment:
There are a lot of "do this stuff n times for these n different algos" IMO this should be written as a function/macro call with a list of algos as argument in all cases. 

I'm not saying that its possible everywhere (small variations/inconsistencies in names, paths etc) but wherever some operation is repeated for several algos, it should be done as a function/macro call with a list of algos.

```
function(for_all_configure_files_ifdef algos)
  foreach(algo ${algos})
    string(TO_UPPER ALGO ${algo})
    configure_file_ifdef(CONFIG_CC310_MBEDTLS_${ALGO}_C
      ${NRF_CC310_MBEDCRYPTO_INCLUDE_PATH}/${algo}_alt.h
      ${generated_include_path}/${algo}_alt.h
      COPYONLY)
  endforeach()
endfunction()

set(algos ccm cmac dhm rsa sha1 sha256 ecp chacha20  ...)
for_all_configure_files_ifdef(${algos})

```

Reply:
Yes. This would clean up a lot. Creating ref: NCSDK-5517